packages/react/src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useId } from 'react-id-generator';
+import { useDidUpdate } from '../../index';
+import TabPanel from './TabPanel';
+import Tab from './Tab';
+
+const [left, right, home, end] = [37, 39, 36, 35];
+
+interface TargetElement extends EventTarget {
+  id?: string;
+}
+
+interface TabsProps {
+  value: number;
+  children: React.ReactNode;
+  handleChange: (newValue: number) => void;
+  id?: string;
+  thin?: boolean;
+  className?: string;
+  ariaLabelForTablist?: string;
+}
+
+const Tabs = ({
+  id: propId,
+  value,
+  children,
+  handleChange,
+  thin,
+  className,
+  ariaLabelForTablist
+}: TabsProps): JSX.Element => {
+  const [activeIndex, setActiveIndex] = useState(value);
+  const [id] = propId ? [propId] : useId(1, 'tabs');
+  const focusedTabRef = useRef<HTMLLIElement>(null);
+
+  const tabs = React.Children.toArray(children).filter(
+    child => (child as React.ReactElement<any>).type === Tab
+  );
+  const panels = React.Children.toArray(children).filter(
+    child => (child as React.ReactElement<any>).type === TabPanel
+  );
+
+  const tabCount = tabs.length;
+
+  const handleClick = (event: React.MouseEvent) => {
+    const eventTarget: TargetElement = event.target;
+    if (eventTarget.id?.includes(id)) {
+      const newIndex = Number(eventTarget.id.split('-')[1]);
+      setActiveIndex(newIndex);
+      handleChange(newIndex);
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    const { which } = event;
+    let newIndex: number = activeIndex;
+
+    switch (which) {
+      case left: {
+        newIndex = activeIndex - 1;
+
+        // circularity
+        if (newIndex === -1) {
+          newIndex = tabCount - 1;
+        }
+        setActiveIndex(newIndex);
+        handleChange(newIndex);
+        break;
+      }
+      case right: {
+        newIndex = activeIndex + 1;
+
+        // circularity
+        if (newIndex === tabCount) {
+          newIndex = 0;
+        }
+        setActiveIndex(newIndex);
+        handleChange(newIndex);
+        break;
+      }
+      case home: {
+        newIndex = 0;
+        setActiveIndex(newIndex);
+        handleChange(newIndex);
+        break;
+      }
+      case end: {
+        newIndex = tabCount - 1;
+        setActiveIndex(newIndex);
+        handleChange(newIndex);
+        break;
+      }
+    }
+  };
+
+  const tabComponents = tabs.map((child, index) => {
+    const { ...other } = (child as React.ReactElement<any>).props;
+    const selected = index === value;
```

Comment:
Shouldn't this be `activeIndex`?
```suggestion
    const selected = index === activeIndex;
```